torch/utils/cpp_extension.py

```diff
@@ -2005,7 +2012,8 @@ def load_inline(name,
     sources = [cpp_source_path]
 
     if cuda_sources:
-        cuda_sources.insert(0, '#include <torch/types.h>')
+        if not no_header:
+            cuda_sources.insert(0, '#include <torch/types.h>')
         cuda_sources.insert(1, '#include <cuda.h>')
```

Comment:
maybe also skip these headers? The `no_headers` actually means that you're responsible for all your includes.